### PR TITLE
feat(logs): adds statistics to report logs

### PR DESCRIPTION
**Pull request by CapKicklee**
Adds statistics for push and pull modules, based on #19 suggestions

### CODE REVIEW

File: internal/app/pull/cli.go [hunk 1]
```diff
@@ -121,6 +134,12 @@ func NewCommand(fullName string, err *os.File, out *os.File, in *os.File) *cobra
 				fmt.Fprintln(err, e3.Error())
 				os.Exit(1)
 			}
+
+			duration := time.Since(startTime)
+			over.MDC().Set("duration", duration)
+			stats := pull.Compute()
+			over.SetGlobalFields([]string{"duration"})
+			log.Info().RawJSON("stats", stats.ToJSON()).Int("return", 0).Msg("End LINO in pull mode")
```
Comment (adrienaury) [inline internal/app/pull/cli.go:142]: I think you should have this code only once in the main command.

The logs could be 

```json
{"msg": "Start LINO", "global flag 1": "value", ...}
{"msg": "Pull mode", "pull flag 1": "value", ...}
{"msg": "End LINO", "stats": "value", "code": 0}
```

**Review by adrienaury — CHANGES_REQUESTED** [review]

File: internal/app/pull/cli.go (hunk 1 above)
Comment (CapKicklee) [reply internal/app/pull/cli.go]: I've made the changes but it implies changing a lot of tests (logs are printed oud in stderr, which means a lot of assertions stating that systemerr should be empty are now false.

Is it the expected behaviour according to you? Should I refactor the code again or should I keep on going and change all impacted tests?

Comment (adrienaury) [reply internal/app/pull/cli.go]: I don't understand why the change implies a different output in stderr. Your initial solution is ok with the "systemerr should be empty" assertion ?
I think the less impact on venom tests the better, but why is it different with this solution ?

Comment (adrienaury) [reply internal/app/pull/cli.go]: Maybe the default log level should be ERROR or WARN, not INFO

Comment (CapKicklee) [reply internal/app/pull/cli.go]: Well I changed the place of the 2 log calls from the run function of each module to the main function from the main module, which prints out "Start LINO" and "End LINO" whenever LINO is executed

Comment (adrienaury) [reply internal/app/pull/cli.go]: Ok i see ! Sorry for the trouble then you can put it back as it was at the beginning. I didn't realize that these logs were existing before this PR

Comment (CapKicklee) [reply internal/app/pull/cli.go]: No worries! Actually, the logs for "Start LINO" and "End LINO" didn't exist before this PR, I added them. First time, I added them in pull and push module, but there were redundant and you were right to ask me to group them into the main module.

Maybe I misunderstood your advice? I put them around the `rootCmd.Execute()` block, but is there a better way?

Comment (adrienaury) [reply internal/app/pull/cli.go]: Yes I was asking if it is possible to have "Start LINO" and "End LINO" everytime not only on pull and push commands.

The stats object can be log only on push and pull command however. In think in other commands, it is ok to not have a report on number of object processed and timing measeure.

Comment (CapKicklee) [reply internal/app/pull/cli.go]: Ok, so I do agree with you, and that's exactly what I did, but this impacts almost every venom tests, because these "Start" and "End"' logs didn't exist before, and thus, we need to refactor those checks on systemerr value.

There's also a check on stats value, and if not present, it will only log "End LINO" without having an empty stats object.

Comment (adrienaury) [reply internal/app/pull/cli.go]: It's because INFO should not be the default level. If you check PIMO currently the defautl log level is ERROR.

If ERROR was the default log level in LINO, no venom test would be impacted, except those with -vinfo, -vdebug or -vtrace (a few of them)

Comment (CapKicklee) [reply internal/app/pull/cli.go]: Oh okay I see! Sorry, I didn't understand this, I will change this and see if it behaves as expected!

Thanks for helping out!

Comment (adrienaury) [reply internal/app/pull/cli.go]: You're welcome and thank YOU for contributing !

Comment (CapKicklee) [reply internal/app/pull/cli.go]: Found a workaround, should be fine now

File: internal/app/pull/cli.go [hunk 2]
```diff
@@ -121,6 +138,12 @@ func NewCommand(fullName string, err *os.File, out *os.File, in *os.File) *cobra
 				fmt.Fprintln(err, e3.Error())
 				os.Exit(1)
 			}
+
+			duration := time.Since(startTime)
+			over.MDC().Set("duration", duration)
+			stats := pull.Compute()
+			over.MDC().Set("stats", stats.ToJSON())
+			over.SetGlobalFields([]string{"duration"})
```
Comment (adrienaury) [inline internal/app/pull/cli.go:146]: Why is this line needed ?

File: pkg/push/driver.go [hunk 3]
```diff
@@ -51,8 +53,12 @@ func Push(ri RowIterator, destination DataDestination, plan Plan, mode Mode, com
 			if errCommit != nil {
 				return errCommit
 			}
+			IncCommitsCount()
 		}
+		IncInputLinesCount()
 	}
+	pushStats := Compute()
+	log.Info().RawJSON("stats", pushStats.ToJSON()).Msg("Push over")
```
Comment (adrienaury) [inline pkg/push/driver.go:61]: Isn't this logged twice ? (line 86 in main.go)

Comment (adrienaury) [reply pkg/push/driver.go]: The problem is also on the pull command

**Comment by adrienaury** [comment]
Thank you 🙏 
I think now you just need to update the CHANGELOG and I will merge

**Comment by CapKicklee** [comment]
Thank you for your help, I updated the CHANGELOG just tell me if something is missing or should be changed!
